api/src/main/java/com/netflix/iceberg/types/Conversions.java

```diff
@@ -98,7 +99,7 @@ public static ByteBuffer toByteBuffer(Type type, Object value) {
         }
       case UUID:
         UUID uuid = (UUID) value;
-        return ByteBuffer.allocate(16).order(ByteOrder.LITTLE_ENDIAN)
+        return ByteBuffer.allocate(16).order(ByteOrder.BIG_ENDIAN)
```

Comment:
Just to be sure, the `putLong` calls are in the right order, correct?

Could you add a test that validates the UUID example from the spec? I want to make sure we get this right.

Reply:
@rdblue I am afraid the order is correct but indices are not.

Here is an example I tested:

```
UUID randomUUID = UUID.randomUUID();
System.out.println(randomUUID.toString());
// abcc5e8d-81f3-4ea2-b624-a14ce8c513b9
ByteBuffer buffer = Conversions.toByteBuffer(UUIDType.get(), randomUUID);
Object value = Conversions.fromByteBuffer(UUIDType.get(), buffer);
System.out.println(value);
// abb624a1-4ce8-c513-b900-000000000000
```

So the logic in `toByteBuffer` should really be:

```
      case UUID:
        UUID uuid = (UUID) value;
        return ByteBuffer.allocate(16).order(ByteOrder.BIG_ENDIAN)
            .putLong(0, uuid.getMostSignificantBits())
            .putLong(8, uuid.getLeastSignificantBits());
```

The test that I added passes because UUIDs are represented as `fixed[16]` in Parquet and we invoke only `fromByteBuffer` in this case.

Do we want to have a separate suite for `Conversions`?

Reply:
Yes, we should probably have one. Sorry about this, it's my bad for not testing properly.

Reply:
No worries, let me extend this PR with a suite for `Conversions`.